### PR TITLE
rm require in core-frontend

### PR DESCRIPTION
**Pull request by aruniverse**
Remove `require` call in core-frontend preventing pure esm usage of the pkg
Like we do with the cjs require, in esm we import a file (package.json) at runtime which doesnt exist at compile time, so we suppress the ts compiler error.
If we import the file where it actually is, it will copy the json file into the lib folder, and the paths of our output will be different (lib/esm/src instead of just lib/esm)

ToDo later:
investigate what other libraries are doing to expose the version
possibilities:
- [ ] have a txt file that has the version string, and do some pre-processing (like we do w/ imodel-native) (personally not a huge fan of this approach)
- [ ] look into [rootDirs](https://www.typescriptlang.org/tsconfig#rootDirs)

### CODE REVIEW

File: core/frontend/src/IModelApp.ts [hunk 1]
```diff
@@ -6,9 +6,12 @@
  * @module IModelApp
  */
 
+// @ts-expect-error resolves from the lib/{cjs,esm} dir
```
Comment (aruniverse) [inline core/frontend/src/IModelApp.ts:9]: does correctly resolve the pkg.json at runtime and will not copy the package.json into the lib/esm or lib/cjs dir at compile time

**Comment by pmconne** [comment]
@markschlosseratbentley @iTwin/itwinjs-core-display does anyone have some time to verify DTA and DPTA still work?

**Comment by markschlosseratbentley** [comment]
> @markschlosseratbentley @iTwin/itwinjs-core-display does anyone have some time to verify DTA and DPTA still work?

I will verify DTA in a bit.
I propose we kick off an Image Tests on this branch to verify DPTA.

Unless that conflicts with @MarcNeely using the Image Tests system today.

**Review by markschlosseratbentley — CHANGES_REQUESTED** [review]
I will approve once we verify DTA and DPTA.

**Comment by markschlosseratbentley** [comment]
DTA still appears to build + run + rebuild + detect code changes.  Using `npm run start:electron` still only shows .js files, but I think that was a previous problem (`npm run start` shows TypeScript).

**Review by markschlosseratbentley — APPROVED** [review]
Image Tests successfully ran with a version of DPTA built off this branch.

**Pull request merged**